community/kernel/src/main/java/org/neo4j/kernel/impl/store/id/IdGeneratorImpl.java

```diff
@@ -372,6 +367,33 @@ public synchronized void close()
         }
     }
 
+    private void closeChannelAndPutThisInstanceIntoClosedState() throws IOException
+    {
+        // flush and close
+        fileChannel.force( false );
+        fileChannel.close();
+        fileChannel = null;
+        // make this generator unusable
+        highId.set( -1 );
+    }
+
+    @Override
+    public synchronized void closeAndSafeDelete()
```

Comment:
Perhaps we could bake this into the current `delete` method instead? So that it would close if necessary, then delete?